README.md

# CLASSIC BOTTOM NAVIGATION BAR💫

By using this package, you will be able to easily create a beautiful bottom navigation bar in flutter and use it on diffrent Os, here is the documantation about how to use it💡

🟢داکیومنت فارسی در مورد نحوه استفاده از این پکیج در ادامه آمده است🟢


![MyVideo_1](https://user-images.githubusercontent.com/86960501/214843370-813e48ae-708d-4793-9270-9a69adb240cd.gif)

![vidtwo](https://user-images.githubusercontent.com/86960501/214843500-1259c718-b8c2-432b-8f9e-71e8972d3d99.gif)




## How to use⁉️

#### set a bottom navigation bar in Scaffold: ✅

```
Scaffold(
      bottomNavigationBar: ClassicButtomNavBar(
        items: const [
          ClassicBottomNavyBarItem(),
          ClassicBottomNavyBarItem(),
          ClassicBottomNavyBarItem(),
          ...
        ],
      ),
    );
```

The arguments needed for (ClassicButtomNavBar) 🟢

| Parameter | Type     | Description                |
| :-------- | :------- | :------------------------- |
| `backgroundColor` | `Color` | **Optional**. color of container |


| Parameter | Type     | Description                       |
| :-------- | :------- | :-------------------------------- |
| `selectedIndex`      | `int` | **Optional**. index of selected page |

| Parameter | Type     | Description                       |
| :-------- | :------- | :-------------------------------- |
| `iconSize`      | `int` | **Optional**. size of icons in bottom nav bar |

| Parameter | Type     | Description                       |
| :-------- | :------- | :-------------------------------- |
| `height`      | `double` | **Optional**. height of the classic bottom nav bar|

| Parameter | Type     | Description                       |
| :-------- | :------- | :-------------------------------- |
| `width`      | `double` | **Optional**. width of the classic bottom nav bar : |

| Parameter | Type     | Description                       |
| :-------- | :------- | :-------------------------------- |
| `borderRadius`      | `double` | **Optional**. the radius of corners of bottom nav bar|

| Parameter | Type     | Description                       |
| :-------- | :------- | :-------------------------------- |
| `items`      | `List<ClassicBottomNavyBarItem>` | **required**. list of items in bottom nav bar |

| Parameter | Type     | Description                       |
| :-------- | :------- | :-------------------------------- |
| `onItemSelected`      | `callbackFunction` | **required**. function when the selecting items|



#### Full example of this package💡

```
Scaffold(
      backgroundColor: const Color.fromARGB(255, 0, 0, 0),
      body: myRoutes[_currentIndex],
      bottomNavigationBar: ClassicButtomNavBar(
        selectedIndex: _currentIndex,
        onItemSelected: (value) => setState(() {
          print("🛑$_currentIndex🛑");
          _currentIndex = value;
          print("🟢$_currentIndex🟢");
        }),
        items: const [
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.home),
            title: "H O M E",
          ),
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.search),
            title: "S E A R C H",
          ),
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.person),
            title: "P R O F I L E",
          ),
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.settings),
            title: "S E T T I N G S",
          ),
        ],
      ),
    );

```






## ⁉️نحوه استفاده 

#### ابتدا ویجت منو مورد نظر را در اسکافولد تعریف کنید: ✅

```
Scaffold(
      bottomNavigationBar: ClassicButtomNavBar(
        items: const [
          ClassicBottomNavyBarItem(),
          ClassicBottomNavyBarItem(),
          ClassicBottomNavyBarItem(),
          ...
        ],
      ),
    );
```

(ClassicButtomNavBar) پارامتر های اجباری و اختیاری موجود برای استفاده در کلاس  🟢

| پارامتر | نوع     | توضیحات                |
| :-------- | :------- | :------------------------- |
| `backgroundColor` | `Color` | **اختیاری**. رنگ پس زمینه منو  |


| پارامتر | نوع     | توضیحات                       |
| :-------- | :------- | :-------------------------------- |
| `selectedIndex`      | `int` | **اختیاری**. شمارنده صفحات انتخاب شده در منو |

| پارامتر | نوع     | توضیحات                       |
| :-------- | :------- | :-------------------------------- |
| `iconSize`      | `int` | **اختیاری**. سایز آیکون های استفاده شده در منو |

| پارامتر | نوع     | توضیحات                       |
| :-------- | :------- | :-------------------------------- |
| `height`      | `double` | **اختیاری**. تنظیم ارتفاع منو |

| پارامتر | نوع     | توضیحات                       |
| :-------- | :------- | :-------------------------------- |
| `width`      | `double` | **اختیاری**. تنظیم عرض منو  : |

| پارامتر | نوع     | توضیحات                       |
| :-------- | :------- | :-------------------------------- |
| `borderRadius`      | `double` | **اختیاری**. تنظیم میزان گرد بودن گوشه های منو|

| پارامتر | نوع     | توضیحات                       |
| :-------- | :------- | :-------------------------------- |
| `items`      | `List<ClassicBottomNavyBarItem>` | **اجباری**. لیست آیتم های  منو مورد  |

| پارامتر | نوع     | توضیحات                       |
| :-------- | :------- | :-------------------------------- |
| `onItemSelected`      | `callbackFunction` | **اجباری**. تابع مورد نیاز هنگام انتخاب هر ایتم|







## Author👦

- [@Taha ahvazi ](https://github.com/TahaAhvazi)


## License

[MIT](https://choosealicense.com/licenses/mit/)


CHANGELOG.md
## 0.0.1
This is the initial release 
## 0.0.2
Update the documantations.
## 0.0.3
add gif and example to documantations.


lib/src/classic_bottom_navigation_bar.dart
import 'dart:math';

import 'package:flutter/material.dart';

class ClassicButtomNavBar extends StatelessWidget {
  final int selectedIndex;
  final int iconSize;
  final double height, width;
  final Color backgroundColor;
  final double borderRadius;
  final double iconBorderRadius;
  final List<ClassicBottomNavyBarItem> items;
  final ValueChanged<int> onItemSelected;
  const ClassicButtomNavBar({
    Key? key,
    this.selectedIndex = 0,
    this.iconSize = 30,
    this.height = 100,
    this.width = 200,
    this.borderRadius = 12.0,
    this.iconBorderRadius = 12.0,
    this.backgroundColor = const Color.fromARGB(255, 20, 0, 53),
    required this.items,
    required this.onItemSelected,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(12.0),
      child: Container(
        height: height,
        width: width,
        decoration: BoxDecoration(
          color: backgroundColor,
          borderRadius: BorderRadius.circular(borderRadius),
        ),
        child: Row(
          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
          children: items.map((item) {
            var index = items.indexOf(item);
            return GestureDetector(
              onTap: () {
                onItemSelected(index);
              },
              child: _myItem(
                iconSize: iconSize,
                item: item,
                isSelected: index == selectedIndex,
              ),
            );
          }).toList(),
        ),
      ),
    );
  }
}

// ignore: camel_case_types
class _myItem extends StatelessWidget {
  final int iconSize;
  final ClassicBottomNavyBarItem item;
  final bool isSelected;

  const _myItem({
    Key? key,
    required this.iconSize,
    required this.item,
    required this.isSelected,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Padding(
          padding: const EdgeInsets.all(8.0),
          child: AnimatedContainer(
            transform:
                isSelected ? Matrix4.rotationY(pi / 8) : Matrix4.rotationY(0.0),
            curve: Curves.easeInCirc,
            decoration: BoxDecoration(
              color: isSelected
                  ? Colors.blue
                  : const Color.fromARGB(255, 164, 68, 255),
              borderRadius: isSelected
                  ? BorderRadius.circular(30.0)
                  : BorderRadius.circular(10.0),
            ),
            height: isSelected ? 60 : 50,
            width: isSelected ? 60 : 50,
            duration: const Duration(milliseconds: 500),
            child: IconTheme(
              data: const IconThemeData(
                color: Colors.white,
                size: 30,
              ),
              child: item.icon,
            ),
          ),
        ),
        Text(
          item.title,
          style: TextStyle(
            fontWeight: FontWeight.bold,
            color: isSelected ? Colors.white : Colors.deepPurpleAccent,
            fontSize: 12,
          ),
        ),
      ],
    );
  }
}

class ClassicBottomNavyBarItem {
  final Icon icon;
  final String title;

  const ClassicBottomNavyBarItem({
    Key? key,
    required this.icon,
    required this.title,
  });
}


lib/example/classic_bottom_navigation_bar_example.dart
import 'package:flutter/material.dart';
import '../classic_bottom_navigation_bar.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        title: 'Flutter Demo',
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: const Home());
  }
}

class Home extends StatefulWidget {
  const Home({Key? key}) : super(key: key);

  @override
  State<Home> createState() => _HomeState();
}

int _currentIndex = 0;
List myRoutes = [
  const Center(
    child: Text(
      "PAGE 1",
      style: TextStyle(
        color: Color.fromARGB(255, 255, 255, 255),
        fontSize: 50,
        fontWeight: FontWeight.bold,
      ),
    ),
  ),
  const Center(
    child: Text(
      "PAGE 2",
      style: TextStyle(
        color: Colors.white,
        fontSize: 50,
        fontWeight: FontWeight.bold,
      ),
    ),
  ),
  const Center(
    child: Text(
      "PAGE 3",
      style: TextStyle(
        color: Colors.white,
        fontSize: 50,
        fontWeight: FontWeight.bold,
      ),
    ),
  ),
  const Center(
    child: Text(
      "PAGE 4",
      style: TextStyle(
        color: Colors.white,
        fontSize: 50,
        fontWeight: FontWeight.bold,
      ),
    ),
  ),
];

class _HomeState extends State<Home> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color.fromARGB(255, 0, 0, 0),
      body: myRoutes[_currentIndex],
      bottomNavigationBar: ClassicButtomNavBar(
        selectedIndex: _currentIndex,
        onItemSelected: (value) => setState(() {
          // ignore: avoid_print
          print("🛑$_currentIndex🛑");
          _currentIndex = value;
          // ignore: avoid_print
          print("🟢$_currentIndex🟢");
        }),
        items: const [
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.home),
            title: "H O M E",
          ),
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.search),
            title: "S E A R C H",
          ),
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.person),
            title: "P R O F I L E",
          ),
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.settings),
            title: "S E T T I N G S",
          ),
        ],
      ),
    );
  }
}


example/lib/main.dart
import 'package:classic_bottom_navigation_bar/classic_bottom_navigation_bar.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        title: 'Flutter Demo',
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: const Home());
  }
}

class Home extends StatefulWidget {
  const Home({Key? key}) : super(key: key);

  @override
  State<Home> createState() => _HomeState();
}

int _currentIndex = 0;
List myRoutes = [
  const Center(
    child: Text(
      "PAGE 1",
      style: TextStyle(
        color: Color.fromARGB(255, 255, 255, 255),
        fontSize: 50,
        fontWeight: FontWeight.bold,
      ),
    ),
  ),
  const Center(
    child: Text(
      "PAGE 2",
      style: TextStyle(
        color: Colors.white,
        fontSize: 50,
        fontWeight: FontWeight.bold,
      ),
    ),
  ),
  const Center(
    child: Text(
      "PAGE 3",
      style: TextStyle(
        color: Colors.white,
        fontSize: 50,
        fontWeight: FontWeight.bold,
      ),
    ),
  ),
  const Center(
    child: Text(
      "PAGE 4",
      style: TextStyle(
        color: Colors.white,
        fontSize: 50,
        fontWeight: FontWeight.bold,
      ),
    ),
  ),
];

class _HomeState extends State<Home> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color.fromARGB(255, 0, 0, 0),
      body: myRoutes[_currentIndex],
      bottomNavigationBar: ClassicButtomNavBar(
        selectedIndex: _currentIndex,
        onItemSelected: (value) => setState(() {
          // ignore: avoid_print
          print("🛑$_currentIndex🛑");
          _currentIndex = value;
          // ignore: avoid_print
          print("🟢$_currentIndex🟢");
        }),
        items: const [
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.home),
            title: "H O M E",
          ),
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.search),
            title: "S E A R C H",
          ),
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.person),
            title: "P R O F I L E",
          ),
          ClassicBottomNavyBarItem(
            icon: Icon(Icons.settings),
            title: "S E T T I N G S",
          ),
        ],
      ),
    );
  }
}
